### PR TITLE
Fixes #31693 - Restore katello::qpid_client

### PR DESCRIPTION
**Pull request by jturel**
This reverts commit 990b7d2c78a31d271064e912760f8dc0fe0a6f2d.

### CODE REVIEW

**Comment by jturel** [comment]
This works for me in testing, seeing if the build passes

**Comment by jturel** [comment]
@ekohl this commit is a revert but maybe it should have its own issue. What do you think?

**Comment by ehelms** [comment]
I wouldn't mind an installer issue linked to the Katello issue from https://github.com/Katello/katello/pull/9088 to track both.

**Comment by jturel** [comment]
> I wouldn't mind an installer issue linked to the Katello issue from [Katello/katello#9088](https://github.com/Katello/katello/pull/9088) to track both.

Done!

File: spec/classes/qpid_client_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe 'katello::qpid_client' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
+
+      if facts[:operatingsystemmajrelease] == '7'
```
Comment (ekohl) [inline spec/classes/qpid_client_spec.rb:8]: A better practice is to not shadow variables. You can also add conditionals to a context.
```suggestion
  on_supported_os.each do |os, facts|
    context "on #{os}", if: os_facts[:operatingsystemmajrelease] == '7' do
      let(:facts) { os_facts }

```

File: manifests/application.pp [hunk 2]
```diff
@@ -18,6 +18,14 @@
   include certs::pulp_client
   include katello::params
 
+  if $facts['os']['release']['major'] == '7' {
```
Comment (ehelms) [inline manifests/application.pp:21]: We can drop this condition now right @jturel ?

File: spec/classes/qpid_client_spec.rb (hunk 1 above)
Comment (jturel) [reply spec/classes/qpid_client_spec.rb]: This check is no longer there!

**Comment by jturel** [comment]
@ekohl @ehelms thanks for the review - updated!

**Review by ehelms — APPROVED** [review]
LGTM

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
       if facts[:operatingsystemmajrelease] == '7'
         it { is_expected.to contain_class('katello::pulp') }
+        it { is_expected.to contain_class('katello::qpid_client') }
```
Comment (ekohl) [inline spec/classes/init_spec.rb:14]: This isn't EL7 specific so in the test it shouldn't be conditional.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: spec/classes/init_spec.rb (hunk 3 above)
Comment (jturel) [reply spec/classes/init_spec.rb:14]: nice catch

Comment (ekohl) [reply spec/classes/init_spec.rb:15]: This looks swapped. I think `katello::qpid` is EL7-only and `katello::qpid_client` is all versions. At least, https://github.com/theforeman/puppet-katello/blob/a0917838fabc8241e3e5e7f678378a2d45cbea54/manifests/init.pp#L174-L178 implies that.

Comment (jturel) [reply spec/classes/init_spec.rb]: Yep :)

File: spec/classes/application_spec.rb [hunk 4]
```diff
@@ -21,12 +21,20 @@ class { 'katello::params':
           it { is_expected.to create_package('tfm-rubygem-katello') }
           it { is_expected.not_to create_package('tfm-rubygem-katello').that_requires('Anchor[katello::candlepin]') }
           it { is_expected.to create_package('rh-postgresql12-postgresql-evr') }
+
```
Comment (ekohl) [inline spec/classes/application_spec.rb:24]: Nit: is this change needed?

**Review by ekohl — APPROVED** [review]
Small nit, otherwise :+1:

File: spec/classes/application_spec.rb (hunk 4 above)
Comment (jturel) [reply spec/classes/application_spec.rb]: Fixed

**Comment by ehelms** [comment]
Thanks @jturel -- merging so I can include in the release
